packages/asset-uploader/src/patch.ts

```diff
@@ -38,4 +42,28 @@ export const patchAssets = async (
   if (deletedAssetIds.length !== 0) {
     deleteAssets({ projectId, ids: deletedAssetIds }, context);
   }
+
+  // add new assets found in patched version
+  const addedAssets: Asset[] = [];
+  for (const [assetId, asset] of patchedAssets) {
+    // skip stubbed assets
+    if (asset === undefined) {
+      continue;
+    }
+    if (assets.has(assetId) === false) {
+      addedAssets.push(asset);
+    }
+  }
+  await prisma.asset.createMany({
+    data: addedAssets.map((asset) => ({
+      id: asset.id,
+      location: asset.location,
+      name: asset.name,
+      size: asset.size,
+      format: asset.format,
+      projectId: asset.projectId,
+      meta: JSON.stringify(asset.meta),
```

Comment:
I worry that we trust all this data from client

Reply:
Our whole data management trust client data. Though at least we validate it with zod.

Reply:
Yeah, and I worry about that :)
Maybe it's ok for now because we want to move fast, but this can cause security issues or just break things. Potential issues I see here:

1. If I know `name` of someone's asset that I don't have access to, I can set `name` of my asset to that name, and get the file
2. I can cause backend to read / delete local files if I set `location` to `LOCAL`
3. I can set wrong `size/format/meta` and cause bugs. This is probably relatively ok, same as submitting somehow broken tree etc., although in case of size, this may allow to bypass size limits if we have them.